javers-core/src/main/java/org/javers/guava/multimap/MultimapChangeAppender.java

```diff
@@ -0,0 +1,169 @@
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+
+import static java.util.Collections.EMPTY_LIST;
+import static org.javers.common.exception.JaversExceptionCode.VALUE_OBJECT_IS_NOT_SUPPORTED_AS_MAP_KEY;
+
+/**
+ * Compares Guava Multimaps.
+ * <br/>
+ *
+ * It's automatically registered, if Guava is detected on a classpath.
+ *
+ * @author akrystian
+ */
+public class MultimapChangeAppender extends CorePropertyChangeAppender implements CustomPropertyComparator<Multimap, MapChange>{
+
+    private final TypeMapper typeMapper;
+    private final GlobalIdFactory globalIdFactory;
+
+    public MultimapChangeAppender(TypeMapper typeMapper, GlobalIdFactory globalIdFactory){
+        this.typeMapper = typeMapper;
+        this.globalIdFactory = globalIdFactory;
+    }
+
+    @Override
+    public MapChange compare(Multimap left, Multimap right, GlobalId affectedId, Property property){
+        if (!isSupportedContainer(property) || (left != null && left.equals(right))){
+            return null;
+        }
+
+        MultimapType multimapType = typeMapper.getPropertyType(property);
+        OwnerContext owner = new PropertyOwnerContext(affectedId, property.getName());
+
+        List<EntryChange> entryChanges = calculateChanges(multimapType, left, right, owner);
+        if (!entryChanges.isEmpty()){
+            renderNotParametrizedWarningIfNeeded(multimapType.getKeyType(), "key", "Map", property);
+            renderNotParametrizedWarningIfNeeded(multimapType.getValueType(), "value", "Map", property);
+            return new MapChange(affectedId, property.getName(), entryChanges);
+        }else{
+            return null;
+        }
+    }
+
+    private List<EntryChange> calculateChanges(MultimapType multimapType, Multimap left, Multimap right, OwnerContext owner){
+        DehydrateMapFunction dehydrateFunction = getDehydrateMapFunction(multimapType);
+        Multimap leftMultimap = multimapType.map(left, dehydrateFunction, owner);
+        Multimap rightMultimap = multimapType.map(right, dehydrateFunction, owner);
+
+        List<EntryChange> changes = new ArrayList<>();
+
+        for (Object commonKey : Multimaps.commonKeys(leftMultimap, rightMultimap)){
+            Collection leftValues = leftMultimap.get(commonKey);
+            Collection rightValues = rightMultimap.get(commonKey);
+
+            Collection difference = difference(leftValues, rightValues);
+            difference.addAll(difference(rightValues, leftValues));
+            if (difference.size() > 0){
+                calculateValueChanges(changes, commonKey, leftValues, rightValues);
+            }
+        }
+        calculateKeyChanges(leftMultimap, rightMultimap, changes);
+        return changes;
+    }
+
+    /**
+     * Difference that handle properly collections with duplicates.
+     */
+    private static Collection difference(Collection first, Collection second){
```

Comment:
To można by przenieść org.javers.common.collections.Collections jako implementacje wspierającą duplikaty.